libr/core/cbin.c

```diff
@@ -2277,6 +2278,12 @@ static int bin_sections(RCore *r, int mode, ut64 laddr, int va, ut64 at, const c
 		r_cons_printf ("Nm Paddr       Size Vaddr      Memsz Perms %sName\n",
                    chksum ? "Checksum          " : "");
 	}
+	r_list_foreach (sections, iter, section) {
```

Comment:
yeah seems buch better now. only small thing: you could do this only when IS_MODE_SET, because otherwise it's computed every time you print things.

Reply:
huh? and where should I store the value of `segments_only`?

> otherwise it's computed every time you print things

2n is still O(n)

Reply:
where you are storing it now. I know 2n is still O(n), but it's still double the time for average sizes.

Reply:
anyway, as said it's a small thing, but I bet it would make @radare happy :-D

Reply:
as you wish d6aae2d9914e9ad40323916ee0cdb0806e75cdc5 ... assumes that sections can't change after loading, but that's probably true currently since `S` has been killed

> I know 2n is still O(n), but it's still double the time for average sizes.

here it's not going to be double the time ... it's going to be more like going from `kn` to `(k+m)n` where `m` is a much smaller value than `k`, since the additional loop is very simple

Reply:
Why did you turn segments_only into a static variable? there's no need for that.

Reply:
oops 447cdd005581cde970d7852943eba07ecf47b10f